modules/dataquery/jsx/index.tsx

```diff
@@ -0,0 +1,262 @@
+                    />;
+            break;
+        case 'DefineFields':
+            content = <DefineFields allCategories={categories}
+                displayedFields={activeCategory.currentDictionary}
+
+                defaultVisits={visits.default_}
+                onChangeDefaultVisits={visits.modifyDefault}
+                allVisits={visits.all}
+
+                module={activeCategory.module}
+                category={activeCategory.category}
+                onCategoryChange={activeCategory.changeCategory}
+
+                selected={selectedFields}
+                setSelected={fieldActions.setFields}
+
+                onFieldToggle={fieldActions.addRemoveField}
+
+                onChangeVisitList={fieldActions.modifyVisits}
+
+                removeField={fieldActions.remove}
+                onAddAll={fieldActions.addMany}
+                onRemoveAll={fieldActions.removeMany}
+                onClearAll={fieldActions.clear}
+
+                fulldictionary={fulldictionary}
+               />;
+            break;
+        case 'DefineFilters':
+            content = <DefineFilters
+                fields={selectedFields}
+                module={activeCategory.module}
+                category={activeCategory.category}
+
+                displayedFields={activeCategory.currentDictionary}
+
+                categories={categories}
+                onCategoryChange={activeCategory.changeCategory}
+
+                addQueryGroupItem={criteriaActions.addQueryGroupItem}
+                removeQueryGroupItem={criteriaActions.removeQueryGroupItem}
+                addNewQueryGroup={criteriaActions.addNewQueryGroup}
+
+                query={query}
+
+                setQuery={criteriaActions.setCriteria}
+
+                getModuleFields={getModuleFields}
+                fulldictionary={fulldictionary}
+
+                mapModuleName={mapModuleName}
+                mapCategoryName={mapCategoryName}
+            />;
+            break;
+        case 'ViewData':
+            content = <ViewData
+                fields={selectedFields}
+                filters={query}
+
+                fulldictionary={fulldictionary}
+
+                onRun={reloadQueries}
+            />;
+            break;
+        default:
+            content = <div>Invalid tab</div>;
+    }
+    return <div>
+        <div>{content}</div>
+        <NextSteps page={activeTab} fields={selectedFields}
+            filters={query}
+            changePage={
+                // FIXME: Validate page is valid
+                (page) => setActiveTab(page)
+        }/>
+    </div>;
+}
+
+declare const loris: any;
+window.addEventListener('load', () => {
+  const element = document.getElementById('lorisworkspace');
+  if (!element) {
+      throw new Error('Missing lorisworkspace');
+  }
+  const root = createRoot(element);
+
+  root.render(
+    <DataQueryApp
+        queryAdmin={loris.userHasPermission('dataquery_admin')}
```

Comment:
is the dataquery_admin permission in LORIS yet? i can't seem to be able to find it. just wanted to check this isn't supposed to be dataquery_view instead

Reply:
It is intended to be admin but seems to be missing from the SQL patch. I'll add it.